### PR TITLE
Apply manual corrections export edm publishing

### PR DESCRIPTION
**Pull request by mbh329**
## motivations

- Resolves issue #33  
- Apply Manually Mapped Geometries (as described [here](https://nyco365.sharepoint.com/:w:/s/NYCPLANNING/itd/edm/EUqg-Rt9CGFEnTZRDcqO4kABtIkkAjdXmcMhW_595MxiEw) and based off the list of [Priority A projects](https://nyco365.sharepoint.com/:x:/s/NYCPLANNING/itd/edm/Ed_gjcfMFtJOrkjJCXzkQvIBuk_bf1tZjrJ8GkFvTGW8hQ). Priority A projects were identified as capital expense projects  (`type_br = e`) and as a site location (`type=site`). Priority B projects are not currently manually mapped but could be in the future.
 
## changes

- Added in two sql files to handle the corrections process: `sql/create_corrrections.sql` which is set up to handle the creation of the table that will ingest all of the submitted projects that have been successfully digitized and `sql/apply_corrections` which applies the manually researched geometries to the identified projects.
- I added the create_corrections to the `01_dataloading` as I think it makes sense to keep all the input datasets into the same shell script. The geom_corrections file goes then goes through the `preprocessing` where it follows the table conventions set up in the repo (e.g. adding `_`, renames `wkt` [geometry column in original table] to `geom`). The corrections are applied in the `03_spatial` step. 
- The `apply_corrections` file takes the raw `corrections_geom` and create a multi geometry (e.g. point -> multipoint) so that records with the same `unique_id` are collected into one single record as a project in the raw `cbbr_submissions` table can have multiple spatial components to it (e.g. a Community board asking for renovations to multiple NYCHA properties). 
  
## notes

- I am not totally set on the prepocessing of the spatial data - currently I bring in the separate spatial files into QGIS, merge like geometries, and then export to csv. From there, I upload the files locally and then copy them into the DB from the local CSV. Once all the manual corrections have been uploaded, I can settle on a final decision as to how to preprocess the data. I focused on this for too long and should have waited for all of the spatial files to be submitted 
- I think it makes sense to export a `corrections_geom` file with the "final" (where the geometries represent multipoints, multipolygons, or multilines as supposed to the "raw" corrections file) version of the corrections file and can add that in. 

## todo

- [ ] Once merged, finish the publishing action and see if that works as expected.

### CODE REVIEW

File: cbbr_build/sql/apply_corrections.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
+-- Create multi geometry if the records has the same unique_id into a CTE table 
+-- and rename the geometry column from wkt to _multipoint_geom to highlight the change in geometry type
+
+-- Create update points corrections table
+
```
Comment (damonmcc) [inline cbbr_build/sql/apply_corrections.sql]: I like the comments in the point section but also like the header comments below like `--- Try Lines`

would be helpful to add `-- Try Points` here and repeat the `-- Create` and `-- Update` comments in the lines and polygon sections

File: cbbr_build/sql/create_corrections.sql [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+--DROP TABLE IF EXISTS corrections_all_geom;
+--CREATE TABLE corrections_all_geom
+--SELECT WKT, unique_id, shape_leng
+--FROM cbbr_point_corrections
+--UNION
+--SELECT WKT, unique_id, Shape_Length
+--FROM cbbr_line_corrections
+--UNION
+--SELECT WKT, unique_id, Shape_Length, Shape_Area
+--FROM cbbr_poly_corrections;
```
Comment (damonmcc) [inline cbbr_build/sql/create_corrections.sql]: commented out and can be deleted?

File: cbbr_build/output/cbbr_submissions.csv
Comment (damonmcc) [inline cbbr_build/output/cbbr_submissions.csv:1]: looks like these files in `output/` were [from the build in 2020](https://github.com/NYCPlanning/db-cbbr/commit/efac8388aab16b6fd3aaa711f0eb5a3e937c5b9c). might be nice to keep these, maybe in a `output/FY2021/` folder?

**Review by damonmcc — COMMENTED** [review]
lgtm, just a couple of thoughts:
- agree with your note about the value of a `corrections_geom` export, and seems reasonable to wait and include that feature in follow-up work that's all about exports/new columns in exports
- to document the results of applying manual corrections, final counts of either the values in `geo_fucnton` or showing a decrease in NULL counts in the final geometry column (compared to without manual corrections) would help confirm the changes work!

**Comment by fvankrieken** [comment]
Things look good to me as well. I am a geospatial newbie so just curious, was it not practical to do the joining programmatically just based on the different formats of user input?

**Comment by mbh329** [comment]
@fvankrieken I thought it might be easier to do some of the preprocessing in QGIS but I could have done it progamatically. the `ogr2ogr` tool was one of the options I thought about using but I was having some initial difficulties getting this to work so I settled on a bit of a roundabout way to do the preprocessing. In our other data products, we typically have a csv file that handles corrections and that was my initial reasoning behind creating the corrections as a csv

File: cbbr_build/sql/apply_corrections.sql (hunk 1 above)
Comment (mbh329) [reply cbbr_build/sql/apply_corrections.sql]: fixed

File: cbbr_build/sql/create_corrections.sql (hunk 2 above)
Comment (mbh329) [reply cbbr_build/sql/create_corrections.sql]: fixed

File: cbbr_build/output/cbbr_submissions.csv
Comment (mbh329) [reply cbbr_build/output/cbbr_submissions.csv:1]: I added them back in and put them in a output/FY2021 folder and they are in the path in the feature branch but I don't see the changes on the PR change file logs. Am I going crazy?

Comment (mbh329) [reply cbbr_build/output/cbbr_submissions.csv:1]: nvm going crazy, see them now

**Review by damonmcc — APPROVED** [review]

**Pull request merged**